mypy/typestate.py

```diff
@@ -29,6 +33,31 @@ class TypeState:
     _subtype_caches = {}  # type: ClassVar[Dict[TypeInfo, Set[Tuple[Instance, Instance]]]]
     _subtype_caches_proper = {}  # type: ClassVar[Dict[TypeInfo, Set[Tuple[Instance, Instance]]]]
 
+    # This contains protocol dependencies generated after running a full build,
+    # or after an update. These dependencies are special because:
+    #   * They are a global property of the program; i.e. some dependencies for imported
+    #     classes can be generated in the importing modules.
+    #   * Because of the above, they are serialized separately, after a full run,
+    #     or a full update.
+    # `proto_deps` can be None if after deserialization it turns out that they are
+    # inconsistent with the other cache files (or an error occurred during deserialization).
+    # A blocking error will be generated in this case, since we can't proceed safely.
+    proto_deps = {}  # type: Optional[Dict[str, Set[str]]]
+
+    # Protocols (full names) a given class attempted to implement.
+    # Used to calculate fine grained protocol dependencies and optimize protocol
+    # subtype cache invalidation in fine grained mode.
+    _attempted_protocols = {}  # type: Dict[TypeInfo, Set[str]]
+    # We also snapshot protocol members of the above protocols.
+    _checked_against_members = {}  # type: Dict[TypeInfo, Set[str]]
+    # TypeInfos that has been type-checked since latest dependency snapshot update.
+    # This is an optimisation for fine grained mode; during a full run we only take
+    # a dependency snapshot at the very end, so this set will contain all type-checked
+    # TypeInfos. After a fine grained update however, we can gather new dependencies only
+    # from few TypeInfos that were type-checked during this update, because these are
+    # the only that can generate new protocol dependencies.
```

Comment:
This comment was a bit unclear. Which dependencies does the comment refer to (of the three kinds)? It's not clear why only type-checked TypeInfos can generate new dependencies, since elsewhere we've established that dependencies are a global property of a program. Please elaborate a bit. What does type-checked mean exactly here? It could plausibly mean, for example, some of 1) triggered `ClsName` 2) triggered `<ClsName>` 3) file containing the class was modified 4) the target containing the class definition was processed -- but I suspect it means none of these.

Style/grammar nits: "from few TypeInfos" -> "from the (typically) few TypeInfos". "are the only" -> "are the only ones"